### ISSUE TITLE
Option "-u" in setsnap.pl

### ISSUE THREAD

**Issue by Arno-Enslin** [adobe-type-tools/perl-scripts#1]
@ Ken

`-u = Upper end of point size range; used only with '-o' option`

I would reserve -u for usage, although an usage option does not exist in case of setsnap.pl.

Edit:

And with regard to

```
sub ShowHelp {
  print STDERR
```

Shouldn't it print to STDOUT?

**Comment by kenlunde** [comment]
@Arno-Enslin: Just FYI, this Perl script is about 10 years old, and was originally a C program from the mid-1990s. I did consider the poaching of the "-u" command-line option, but given that the script supports only the "-h" command-line option for help/information, I figured that it wouldn't make much difference. About outputting _ShowHelp_ to STDERR or STDOUT, given that the script terminates, does it make much of a difference?

**Comment by Arno-Enslin** [comment]
> does it make much of a difference?

Not really. I have recognized this, because I am redirecting the help of the scripts, contained in the AFDKO and FontTools to text files.

In case of setsnap.pl this:
`if [%1]==[] %setsnapCom% -h > tempSetsnap-h.txt&start tempSetsnap-h.txt&ping 127.0.0.1 -n 3 >NUL&del tempSetsnap-h.txt&exit`

results in a 0 byte file, because the help goes to STDERR only.

So I have to write that:
`if [%1]==[] %setsnapCom% -h 2> tempSetsnap-h.txt&start tempSetsnap-h.txt&ping 127.0.0.1 -n 3 >NUL&del tempSetsnap-h.txt&exit`

With regard to most of the other files with a help, that are contained in the AFDKO, this **(2>)** is **inconsistent.** That's all.

(I am redirecting the help, because in a text editor like Notepad [the standard text file editor on Windows] it is easier on the eyes than in the command line window. The text file becomes deleted three seconds after creation. That is the time, which Notepad needs for opening it.)

[More info about redirection in batch files.](http://ss64.com/nt/syntax-redirection.html)

**Comment by Arno-Enslin** [comment]
By the way: Although Perl is only needed for your scripts in the AFDKO. But wouldn't it be nice, if it also would be integrated in the prebuilt versions of the AFDKO? Or alternatively: I don't know, if this is possible, but maybe your scripts could be converted to Python scripts.

Edit: I just see, that my Perl installation folder is big. Round about 40 MB compressed. Maybe a bit too much for a few scripts only.

**Comment by kenlunde** [comment]
I just pushed a new version of setsnap.pl that adjusts the command-line options, such as supporting both -u (displays only the command-line options) and -h (same as -u but shows more information), and changing -l (lower) and -u (upper) to -b (bottom) and -t (top), respectively. I also changed STDERR to STDOUT.

Actually, an older version (that lacks any headers) of setsnap.pl is included in both [AFDKO](https://www.adobe.com/devnet/opentype/afdko.html) and its [sources](https://github.com/adobe-type-tools/afdko/blob/master/FDK/Tools/SharedData/FDKScripts/setsnap.pl). I'll need to get those updated to this version.

Also, I was raised with Perl, and was crafting Perl scripts that worked with our font-development tools long before AFDKO existed. Pre-AFDKO, our tools ran only on SunOS. Rewriting in Python is not really an option unless someone else wants to do the work. I cannot sling Python at all.
